addons/binding/org.openhab.binding.dwdunwetter/src/main/java/org/openhab/binding/dwdunwetter/internal/data/DwdWarningsData.java

```diff
@@ -0,0 +1,296 @@
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import javax.measure.quantity.Length;
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+import javax.xml.stream.events.XMLEvent;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.cache.ExpiringCache;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.ImperialUnits;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Contains the Data for all retrieved warnings for one thing.
+ *
+ * @author Martin Koehler - Initial contribution
+ */
+public class DwdWarningsData {
+
+    private static final int MIN_REFRESH_WAIT_MINUTES = 5;
+
+    private final Logger logger = LoggerFactory.getLogger(DwdWarningsData.class);
+
+    private List<DwdWarningData> gemeindenData = new LinkedList<>();
+
+    private DwdWarningCache cache = new DwdWarningCache();
+
+    private ExpiringCache<String> dataAccessCached;
+
+    private DateTimeFormatter formatter = new DateTimeFormatterBuilder()
+            // date/time
+            .append(DateTimeFormatter.ISO_LOCAL_DATE_TIME)
+            // offset (hh:mm - "+00:00" when it's zero)
+            .optionalStart().appendOffset("+HH:MM", "+00:00").optionalEnd()
+            // offset (hhmm - "+0000" when it's zero)
+            .optionalStart().appendOffset("+HHMM", "+0000").optionalEnd()
+            // offset (hh - "Z" when it's zero)
+            .optionalStart().appendOffset("+HH", "Z").optionalEnd()
+            // create formatter
+            .toFormatter();
+
+    public DwdWarningsData(String cellId) {
+        DwdWarningDataAccess dataAccess = new DwdWarningDataAccess();
+        this.dataAccessCached = new ExpiringCache<>(Duration.ofMinutes(MIN_REFRESH_WAIT_MINUTES),
+                () -> dataAccess.getDataFromEndpoint(cellId));
+    }
+
+    private String getValue(XMLEventReader eventReader) throws XMLStreamException {
+        XMLEvent event = eventReader.nextEvent();
+        return event.asCharacters().getData();
+    }
+
+    private BigDecimal getBigDecimalValue(XMLEventReader eventReader) throws XMLStreamException {
+        XMLEvent event = eventReader.nextEvent();
+        try {
+            return new BigDecimal(event.asCharacters().getData());
+        } catch (NumberFormatException e) {
+            logger.warn("Exception while parsing a BigDecimal", e);
+            return BigDecimal.ZERO;
+        }
+    }
+
+    private long getTimestampValue(XMLEventReader eventReader) throws XMLStreamException {
```

Comment:
Wouldn't it be nicer to store all timestamps directly as `Instant`, instead of converting to epoch offsets and back?

Reply:
Yes. I have sometimes my problems with Date/Time APIs of Java (What to use when...)

Reply:
The equivalent to a plain timestamp would be a `java.time.Instant`